src/channel/state_broadcast.rs

```diff
@@ -244,12 +244,21 @@ where
         CloseStatus::NewlyClosed
     }
 
+    fn try_receive(&mut self, state_id: StateId) -> Option<(StateId, T)> {
+        let val = self.value.as_ref()?;
```

Comment:
This is pretty much the same code as inside the `receive_or_register`, so that one could have called this. But since it is only 5 lines it doesn't really matter.